torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -461,6 +465,31 @@ def __init__(
             if param not in ignored_params and not isinstance(param, FlatParameter):
                 params.append(param)
 
+        # At this point we are wrapping an atomic FSDP unit. Check module shapes
+        # and broadcast module from rank 0 to ensure everything starts off with
+        # the same parameters.
+        # Hack: move to compute device and back if it is CPU
+        prev_device = next(module.parameters()).device
+        was_cpu = (prev_device == torch.device("cpu"))
+        if was_cpu:
+            warnings.warn(
+                f"Module is input on CPU, we are moving to to {self.compute_device} to perform parameter verification and will move it back after. In the future please specify device_ids which will explicitly tell FSDP to place the module on a particular device."
```

Comment:
Is `self.compute_device` guaranteed to be a CUDA device (and not CPU)?

Reply:
line too long, shall we break it into shorter ones?

Reply:
@awgu Yep it is